projects/vdk-heartbeat/tests/vdk/internal/test_job_controller.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from unittest.mock import patch
+
+import pytest
+from vdk.internal.heartbeat.config import Config
+from vdk.internal.heartbeat.job_controller import JobController
+
+
+@patch.object(JobController, "_execute")
+def test_check_job_execution_finished(patched_method):
+    test_config = Config()
+    test_config.control_api_url = None
+    test_config.vdkcli_oauth2_uri = None
+
+    patched_method.return_value = (
+        '[{"status": "finished", "end_time": ' '"2022-03-01"}]'
+    )
+
+    test_controller = JobController(test_config)
+    test_controller.config.RUN_TEST_TIMEOUT_SECONDS = 10
+
+    res = test_controller.check_job_execution_finished()
+    assert res == "finished"
+
+
+@patch.object(JobController, "_execute")
+def test_check_job_execution_finished_exceptions(patched_method):
+    test_config = Config()
+    test_config.control_api_url = None
+    test_config.vdkcli_oauth2_uri = None
+
+    patched_method.return_value = '[{"status": "finished", "end_time": "None"}]'
```

Comment:
Shouldn't it return None without "" as https://www.pythontutorial.net/advanced-python/python-none 

Can the API return anything else but valid string timestamp or None (e.i. no such value)

Reply:
I believe the result of the request is always in the form of a string, even with 'None', as otherwise, it would raise a JsonDecoderError while converting the string to dictionary on line 363 https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-heartbeat/src/vdk/internal/heartbeat/job_controller.py#L363

Reply:
That does not make sense to me. 

 The API must return either 
```
{"status": "finished"}
```

where the "end_time" key is not present 

```
{"status": "finished", "end_time": "1988-11-12..."}
```
where end_time is valid timestamp . 

Are you saying the API (or vdk-control-cli) returns something else . Anything else seems like a bug either in vdk-control-cli or in the API...

Reply:
Well, if the format was `{"status": "finished"}`, then it would not throw a ValueError, but rather a KeyError. It looks like sometimes the format is `{"status": "finished", "end_time": "<some-value>"}`, where `<some-value>` is not a date string.

Reply:
can you give an example?